src/sparseml/pytorch/torchvision/train.py

```diff
@@ -537,6 +536,38 @@ def collate_fn(batch):
     print(f"Training time {total_time_str}")
 
 
+def _create_model(args, num_classes):
```

Comment:
Would it make more sense to have individual params? or maybe a class here? right now this function needs to know the contents of args namespace, which could be difficult to debug later

Reply:
yeah i can separate it out

Reply:
Updated logic - moved rank management outside this function which should make it easier to debug/test